### ISSUE TITLE
RocksDB state backend causes segfault on Alpine images

### ISSUE THREAD

**Issue by patricklucas** [docker-flink/docker-flink#14]
Alpine Linux uses MUSL instead of glibc, but the version of RocksDB embedded in Flink apparently requires some absent glibc functionality.

```
java.lang.UnsatisfiedLinkError: /tmp/rocksdb-lib-33e471f8d228c175dfc6148869213083/librocksdbjni-linux64.so: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /tmp/rocksdb-lib-33e471f8d228c175dfc6148869213083/librocksdbjni-linux64.so)
```

By installing the Alpine Linux package `libc6-compat`, this error was resolved but was replaced by this one:

```
java.lang.UnsatisfiedLinkError: /tmp/rocksdb-lib-fea5dea17151b5dfdc87d323576a04c3/librocksdbjni-linux64.so: Error relocating /tmp/rocksdb-lib-fea5dea17151b5dfdc87d323576a04c3/librocksdbjni-linux64.so: __strtod_internal: symbol not found
```

I didn't find an obvious solution to that one, but it's possible installing an additional package for compatibility would work.

We might want to consider dropping the official -alpine images until this is resolved.

cc @iemejia @StephanEwen

**Comment by patricklucas** [comment]
Additionally, some `ldd` output:

```
bash-4.3# ldd /tmp/rocksdb-lib-017af084e60916b3bffbf90d411ad14d/librocksdbjni-linux64.so
	ldd (0x56488fb47000)
	libpthread.so.0 => ldd (0x56488fb47000)
	librt.so.1 => ldd (0x56488fb47000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x7fc11320b000)
	libm.so.6 => ldd (0x56488fb47000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7fc112ff9000)
	libc.so.6 => ldd (0x56488fb47000)
	ld-linux-x86-64.so.2 => /lib/ld-linux-x86-64.so.2 (0x7fc112d6c000)
Error relocating /tmp/rocksdb-lib-017af084e60916b3bffbf90d411ad14d/librocksdbjni-linux64.so: __strtod_internal: symbol not found
Error relocating /tmp/rocksdb-lib-017af084e60916b3bffbf90d411ad14d/librocksdbjni-linux64.so: backtrace: symbol not found
Error relocating /tmp/rocksdb-lib-017af084e60916b3bffbf90d411ad14d/librocksdbjni-linux64.so: backtrace_symbols: symbol not found
```

**Comment by StephanEwen** [comment]
I am +1 for dropping the alpine images in favor of images which have the necessary libraries.

How much larger is the ubuntu image in comparison to the alpine?

**Comment by patricklucas** [comment]
The latest builds are 253MB for hadoop27/scala_2.11/debian and 185MB for hadoop27/scala_2.11/alpine.

Because of how large a Flink distribution is, those 68MB have less of an impact than they might for other projects, but a potential 27% decrease still isn't trivial.

Nevertheless, I'm also in favor of dropping the alpine images for now. @iemejia?

**Comment by iemejia** [comment]
Good catch @patricklucas. I agree with the removal, is better to have a reliable image even at the cost of the size, I can take a look for a fix, but I am quite booked until tomorrow at 15h CEST so I will check it after that, so it is up to you if we can wait. What did you do exactly to reproduce the issue just checkpointed ?

**Comment by patricklucas** [comment]
@iemejia: yeah, pretty much. Steps to reproduce:

* `$ docker run --rm -t -i plucas/flink:alpine bash`
* `# echo "state.backend: rocksdb" >> conf/flink-conf.yaml`
* `# echo "state.backend.fs.checkpointdir: file:///tmp/checkpoints" >> conf/flink-conf.yaml`
* `# jobmanager.sh start-foreground local &`
* `# flink run examples/streaming/WindowJoin.jar`

**Comment by iemejia** [comment]
I think I found the fix for this. PTAL at https://github.com/docker-flink/docker-flink/pull/22

**Comment by patricklucas** [comment]
Fixed in #22
